perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java

```diff
@@ -1434,22 +1434,27 @@ public RichMember sponsorMember(PerunSession session, Member sponsored, User spo
 
 	@Override
 	public List<RichMember> getSponsoredMembers(PerunSession sess, Vo vo) throws PrivilegeException, VoNotExistsException {
+		//Filter members based on authorization
+		return getAllSponsoredMembers(sess, vo).stream()
+			.filter(member -> AuthzResolver.authorizedInternal(sess, "filter-getSponsoredMembers_Vo_policy", member, vo))
+			.collect(Collectors.toList());
+	}
+
+	@Override
+	public List<RichMember> getAllSponsoredMembers(PerunSession sess, Vo vo) throws PrivilegeException, VoNotExistsException {
 		Utils.checkPerunSession(sess);
 		Utils.notNull(vo, "vo");
 
 		perunBl.getVosManagerBl().checkVoExists(sess, vo);
 
 		//Authorization
 		if(!AuthzResolver.authorizedInternal(sess, "getSponsoredMembers_Vo_policy", vo)) {
```

Comment:
This policy should be used in the method `getSponsoredMembers`. For this method, you should create a new one.

Reply:
Fixed.